articles/azure/azs-how-create-service-principals.md

```diff
@@ -0,0 +1,97 @@
+
+## Getting started
+
+Use the steps in this article as a guide to:
+
+* Create a service principal for your app.
+* Register your app and create an authentication key.
+* Assign your app to a role.
+
+## Create a service principal for Azure AD
+
+If your Azure Stack uses Azure AD as the identity store, you can create a service principal using the same steps as in Azure, using the Azure portal. Please check that you have the require Azure AD permissions before trying to create a service principle.
+
+### Creating a service principal
+
+To create a service principal for your application:
+
+1. Sign in to your Azure Account through the Public Azure portal.
+2. Select **Azure Active Directory** > **App registrations** > **Add**.
+    ![Microsoft Azure CSP invitation email](images/azs-portal-new-principle1.png)
+
+3. Provide a name and URL for the application. Select either **Web app / API** or **Native** for the type of application you want to create. After setting the values, select **Create**.
+
+    ![Microsoft Azure CSP invitation email](images/azs-portal-new-principle2.png)
+
+### Get Credentials
+
+When logging in programmatically, use the ID for your application and an authentication key. To get these values:
+
+1. From App registrations in Active Directory, select your application.
+
+2. Copy the Application ID and store it in your application code. The applications in the sample applications use client id when referring to the Application ID.
+
+    ![Microsoft Azure CSP invitation email](images/azs-portal-new-principle3.png)
+3. To generate an authentication key, select Keys.
+
+    ![Microsoft Azure CSP invitation email](images/azs-portal-new-principle4.png)
+4. Provide a description of the key, and a duration for the key. When done, select Save.
+
+    ![Microsoft Azure CSP invitation email](images/azs-portal-new-principle5.png)
+5. After you save the key, the key VALUE is displayed. Write down this value because you can't retrieve the key later. Store the key value where your application can retrieve it.
+
+## Assign the service principal to a role
+
+To access resources in your subscription, you must assign the application to a role. Decide which role represents the right permissions for the application.
+
+ > [!TIP]
+ > You can set a role's scope at the level of a subscription, a resource group, or a resource. Permissions are inherited to lower levels of scope. For example, an app with the Reader role for a resource group means that the app can read any of the resources in the resource group.
+
+Use the following steps as a guide for assigning a role to a service principal.
+
+1. In the Azure Stack portal, navigate to the level of scope you want to assign the application to. For example, to assign a role at the subscription scope, select Subscriptions.
+
+2. Select the subscription to assign the application to. In this example, the subscription is highlighted.
```

Comment:
Don't think you need the second sentence.

Reply:
done